pkg/apis/kops/cluster.go

```diff
@@ -807,10 +807,6 @@ func (c *Cluster) FillDefaults() error {
 		c.Spec.MasterInternalName = "api.internal." + c.ObjectMeta.Name
 	}
 
-	if c.Spec.MasterPublicName == "" {
```

Comment:
I am not a big fan of this method. E.g if the API name fields should be set in new_cluster or in populate cluster spec or something when dns type is none. Some care needs to be taken here wrt all the paths that may create a cluster object that expects this being set here.

I could not see that we have validation ensuring this fiels is always set.

Reply:
Not quite sure why someone would expect this to be there. It should not be populated to private clusters or Gossip ones either.